lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemDataSet.cs

```diff
@@ -0,0 +1,116 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.Hadoop;
+using Org.Apache.REEF.IO.FileSystem.Local;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+
+namespace Org.Apache.REEF.IO.PartitionedData.FileSystem
+{
+    internal sealed class FileSystemDataSet<T> : IPartitionedDataSet
+    {
+        private readonly Dictionary<string, IPartitionDescriptor> _partitions;
+        private readonly int _count ; 
+        
+        [Inject]
+        private FileSystemDataSet(
+            [Parameter(typeof(AllFilePaths))] ISet<string> filePaths,
+            [Parameter(typeof(TypeOfFileSystem))] string fileSystemType, 
```

Comment:
Have implemented HadoopFileSystemProvider in REEF-803. Here we will inject IFileSystem directly. 
If the user would like to use HadoopFileSystem, he should add HadoopFileSystemConfig. Otherwise, default file system LocalFileSystem will be used.